### PR TITLE
[WORKING DRAFT] Better dungeon door detection/overlay

### PR DESCRIPTION
**Pull request by GatienDoesStuff**
I've just spent the day rewriting the code that detects dungeons doors.

Let me know if I'm doing anything wrong / there are any concerns regarding the proposed code.

# Summary of changes compared to current implementation
- Cleanup of now unused functions
- Doors towards the blood room are now detected by looking at the map, allowing them to be rendered no matter where the player currently is in the dungeon (just like `BetterMap` does on 1.8)
- This also means that the fairy room is now highlighted, which wasn't the case before
- Fixes bugs related to the key being picked up naturally after it times out (the door wouldn't get recolored to green)

# Concerns
- Code quality ?
- Hardcoded values for map colors, as well as position code that might not be easily understood
- Currently, the doors are checked for every tick, which might not be a good thing for performance. It could be possible to only check when necessary :
  - Upon starting a run
  - Upon opening a door
  - Upon fairy room being discovered (harder to detect)
  
  Feel free to test those changes, things have been working pretty smoothly on my end, and I haven't been able to find any bugs / wrong placements for the doors.

### CODE REVIEW

**Comment by GatienDoesStuff** [comment]
Just did a dozen runs with this, and I haven't really found any issues, doors are rendering where they're expected to

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Looks good, just some small things and formatting.

File: src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/Room.java [hunk 1]
```diff
@@ -416,7 +395,8 @@ protected boolean checkBlock(ClientWorld world, BlockPos pos) {
             Scheduler.INSTANCE.schedule(() -> matchState = MatchState.MATCHING, 50);
             reset();
             return true;
-        } else if (matchingRoomsSize == 1) {
+        }
+        else if (matchingRoomsSize == 1) {
```
[the remark this answers is not in the document]
Comment (GatienDoesStuff) [reply src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/Room.java:399]: My bad on those, I didn't correctly apply the formatting config listed on the wiki.

While the other changes all apply just like you did, it seems like the formatter really doesn't want to have the `else if` on the same line as the closing bracket.
Is it a config I'm missing ? If I apply this, the next person to work on that file might just revert it.

File: src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/DungeonManager.java [hunk 2]
```diff
@@ -770,5 +774,54 @@ private static void reset() {
...
+
+        MapState map = FilledMapItem.getMapState(DungeonMap.getMapIdComponent(client.player.getInventory().main.get(8)), client.world);
+        if (map == null || mapEntrancePos == null || mapRoomSize == 0) {
+            LOGGER.error("[Skyblocker Dungeon Secrets] Failed to get map state.");
+            return;
+        }
+
+        Vector2i nWMostRoom = getMapPosForNWMostRoom(mapEntrancePos, mapRoomSize);
+
+        for (int x = nWMostRoom.x + mapRoomSize / 2; x < 127; x += mapRoomSize + 4) {
```
[the remark this answers is not in the document]
Comment (GatienDoesStuff) [reply src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/DungeonManager.java:801]: You're right, this went over my head.

**Comment by GatienDoesStuff** [comment]
I believe this should be it, minus the small nitpick about `else if` above.

File: src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/Room.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/dungeon/secrets/Room.java:398]: I’m not sure. The intellij default keeps them on the same line.

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**